partner_tier_validation/models/res_partner.py

```diff
@@ -48,3 +48,21 @@ def write(self, vals):
         if "stage_id" in vals and vals.get("stage_id") in self._state_from:
             self.restart_validation()
         return res
+
+    def _evaluate_valid_tiers(self):
+        tiers = self.env["tier.definition"].search(
+            [
+                ("model", "=", self._name),
+                ("company_id", "in", [False] + self.env.company.ids),
+            ]
+        )
+        return any([self.evaluate_tier(tier) for tier in tiers])
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        default_stage = self._get_default_stage_id().id
+        for partner in res:
+            if partner._evaluate_valid_tiers() and partner.stage_id != default_stage:
+                partner.stage_id = default_stage
+        return res
```

Comment:
I don't totally understand the purpose of  `_evaluate_valid_tiers`.

The code also prevent creating Partners in a non default stage. For example that prevents importing partenrs as alerady confirmed. Is this a good idea?

I suggest simplifying and preprocess the create values:
```suggestion
        default_stage_id = self._get_default_stage_id().id
        for vals in vals_list:
            if not vals.get("stage_id") and default_stage_id:
                vals["stage_id"] = default_stage_id
        return super().create(vals_list)
```

Reply:
Hi, Thank you for your review. 
Indeed, your proposition is simplier and I applied it.